app/src/main/java/com/project/ianime/screens/viewanime/AnimeDetailFragment.kt

```diff
@@ -78,26 +78,29 @@ class AnimeDetailFragment : BaseFragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+
         loadAnimeDetails()
     }
 
     private fun loadAnimeDetails() {
-        animeViewModel.getAnimeById(animeTargetId)?.let { targetAnime ->
-            // set title to each anime name
-            actionBarService.setTitle(targetAnime.animeName, toolbar)
-
-            // load data details
-            animeName.text = targetAnime.animeName
-            targetAnime.animeImageUrl?.let {
-                imageUtils.loadImageFromDisk(it, animeProfile)
+        animeViewModel.getAnimeById(animeTargetId) { targetAnime ->
+            targetAnime?.let {
+                // set title to each anime name
+                actionBarService.setTitle(it.animeName, toolbar)
+
+                // load data details
+                animeName.text = it.animeName
+                targetAnime.animeImageUrl?.let {
+                    imageUtils.loadImageFromDisk(it, animeProfile)
+                }
+                animeType.text = getString(R.string.anime_type_title, getString(it.type.label))
+                animeCountry.text = getString(R.string.anime_country_title, getString(it.country.label))
+                val releaseYear = it.releaseYear ?: getString(R.string.not_applicable_data)
+                animeReleasedYear.text = getString(R.string.anime_release_year_title, releaseYear)
+                animeStatus.text = getString(R.string.anime_status_title, getString(it.status.label))
+                val animeDescription: String = it.synopsis ?: getString(R.string.empty_description_message)
+                animeIntro.text = animeDescription
```

Comment:
do we need to consider empty value here?

Reply:
see [add anime not found scenario](https://github.com/AM-Kitty/iAnime/pull/10/commits/694137e437fade69a0bf36cbb9add7e0fd2a3dd8)